src/node.cpp

```diff
@@ -1831,6 +1861,13 @@ LocalNode* LocalNode::unserialize(Sync* sync, string* d)
     m_off_t size = MemAccess::get<m_off_t>(ptr);
     ptr += sizeof(m_off_t);
 
+    const bool hasExtensionBytes = size == LOCALNODE_EXTENSION_BYTES_FLAG;
```

Comment:
I think `LocalNode::unserialize` is only called from `Sync::readstatecache`, and it passes an independent string from the database each time.   The old code does not check if it got to the end of the string, so we could just append new fields, and older SDK versions will ignore them.   I think we can get away without the extension bytes flag, and retain backward compatibility?  thanks

Reply:
We could do that if `mtime` was serialized the same way as other fields with a fixed number of bytes. However, `Serialize64::serialize` only writes out as many bytes as it needs so if we appended to the end then we wouldn't know when `mtime` ends and our extension begins?